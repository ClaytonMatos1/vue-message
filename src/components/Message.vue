<template>
    <li class="message" @click="handleClick">{{ message }}</li>
</template>

<script>
    export default {
        props : {
            message : {
                type : String,
                required : true,
                validator : message => message.length > 1
            },
            author : {
                type : String,
                default : 'Jos√©'
            }
        },

        methods : {
            handleClick() {
                this.$emit('message-clicked', this.message);
            }
        },

        created () {
            console.log('Message.vue CREATED!');
        }
    }
</script>

<style scoped>
    .message {
        margin-top: 10px;
    }
</style>
