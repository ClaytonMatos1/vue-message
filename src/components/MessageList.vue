<template>
    <ul>
        <message
			@message-clicked="handleMessageClick"
			:message="message"
			v-for="message in messages"
			:key="message"/>
    </ul>
</template>

<script>
	import Message from './Message';

    export default {
        name : 'message-list',

		props : ['messages'],

		methods : {
			handleMessageClick (message) {
				console.log(message);
			}
		},

		components : {
			Message
		}
    }
</script>
