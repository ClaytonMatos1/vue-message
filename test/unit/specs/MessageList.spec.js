import { mount }   from "vue-test-utils";
import MessageList from '@/components/MessageList';
import Message     from '@/components/Message';

describe('MessageList', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(MessageList, {
            propsData : {
                messages: ['Adeus']
            }
        });
    });

    it('has received ["Adeus"] as the message property', () => {
        expect(wrapper.vm.messages).toEqual(['Adeus']);
    });

    it('has the expected html structure', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('is a MessageList component', () => {
        expect(wrapper.is(MessageList)).toBeTruthy();
        expect(wrapper.is('ul')).toBeTruthy();
    });

    it('contains a Message component', () => {
        expect(wrapper.contains(Message)).toBeTruthy();
        expect(wrapper.contains('.message')).toBeTruthy();
    });

    it('both MessageList and Message are vue instances', () => {
        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.find(Message).isVueInstance).toBeTruthy();
    });

    it('message element exists', () => {
        expect(wrapper.find('.message')).toBeTruthy();
    });

    it('message is not empty', () => {
        expect(wrapper.find(Message).isEmpty()).toBeFalsy();;
    });

    it('message has a class attribute set to "message"', () => {
        expect(wrapper.find(Message).attributes('class', 'message')).toBeTruthy();
    });

    it('message component has the .message class', () => {
        expect(wrapper.find(Message).classes()).toContain('message');
    });

    it('message component has style margin-top equal 10px', () => {
        // expect(wrapper.find(Message).hasStyle('margin-top', '10px')).toBe(true)
    });
});