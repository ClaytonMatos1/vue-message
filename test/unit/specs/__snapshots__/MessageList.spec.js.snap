// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList has the expected html structure 1`] = `
<ul>
  <li
    class="message"
  >
    Adeus
  </li>
</ul>
`;
